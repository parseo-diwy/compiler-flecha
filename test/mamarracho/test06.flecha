
def main =
  let x = 44 in
  (
    unsafePrintInt x; unsafePrintChar '\n';
    let y = x in
    let x = 43 in
    (
      unsafePrintInt x; unsafePrintChar '\n';
      unsafePrintInt y; unsafePrintChar '\n';
      let z = y in
      let y = x in
      let x = 42 in (
        unsafePrintInt x; unsafePrintChar '\n';
        unsafePrintInt y; unsafePrintChar '\n';
        unsafePrintInt z; unsafePrintChar '\n'
      );
      unsafePrintInt x; unsafePrintChar '\n';
      unsafePrintInt y; unsafePrintChar '\n'
    );
    unsafePrintInt x; unsafePrintChar '\n'
  )

