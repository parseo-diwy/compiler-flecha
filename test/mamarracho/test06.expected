44
43
44
42
43
44
43
44
44
